### ISSUE TITLE
JavaTokenParsers.ident doesn't work on Scala.js

### ISSUE THREAD

**Issue by rhz** [scala/scala-parser-combinators#55]
Hi, I'm using `libraryDependencies += "org.scala-js" %%% "scala-parser-combinators" % "1.0.2"` in build.sbt and `JavaTokenParsers.ident` doesn't work for `"a"` (it works in the console but not in the browser). The error message is

```
[1.1] failure: string matching regex `\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*' expected but `a' found

a
^
```

The code to replicate it:

```scala
val p = new JavaTokenParsers {}
println(p.parse(p.ident,"a"))
```

Cheers,
Ricardo

**Comment by dcsobral** [comment]
This is defined as

    """\p{javaJavaIdentifierStart}\p{javaJavaIdentifierPart}*""".r

I presume all regex used by Scala.js is Javascript regex, which is rather different from Java regex, so this, and possible `stringLiteral` as well, doesn't work.

**Comment by rhz** [comment]
In that case wouldn't it be better to remove JavaTokenParsers from the API in Scala.js?

**Comment by aniketbhatnagar** [comment]
+1 Its a bit disappointing that out of the box parsers that use java Regex may not work in scalajs.

**Comment by aniketbhatnagar** [comment]
This may be related to/gated by https://github.com/scala-js/scala-js/issues/1201

**Comment by gourlaysama** [comment]
This was fixed in master in #76, so it will end up in v1.1.0, which shouldn't be long now.
